scipy/optimize/_lsq/lsq_linear.py

```diff
@@ -0,0 +1,456 @@
+        (that is, whether a variable is at the bound):
+
+            *  0 : a constraint is not active.
+            * -1 : a lower bound is active.
+            *  1 : an upper bound is active.
+
+        Somewhat arbitrary because it is determined within a tolerance
+        threshold.
+    nit : int
+        Number of iterations. Zero if the unconstrained solution is optimal.
+    status : int
+        Reason for algorithm termination:
+
+            * -1 : the algorithm was not able to make progress on the last
+              iteration.
+            *  0 : the maximum number of iterations is exceeded.
+            *  1 : the uniform norm of the scaled gradient is less than `tol`.
+            *  2 : the relative change of the cost function is less than `tol`.
+            *  3 : the unconstrained solution is optimal.
+
+    message : str
+        Verbal description of the termination reason.
+    success : bool
+        True if one of the convergence criteria is satisfied (`status` > 0).
+
+    See Also
+    --------
+    nnls : Linear least-squares with non-negativity constraint
+
+    Notes
+    -----
+    The algorithm first computes the unconstrained solution by
+    `numpy.linalg.lstsq` or `scipy.sparse.linalg.lsmr` depending on `solver`.
+    This solution is returned as optimal if it fits within the bounds.
+
+    The algorithm for essentially constrained problems is based on paper [1]_.
+    The algorithm tries to solve system :math:`D^2 g = 0`, where
+    :math:`g = A^T (A x - b)` is the gradient of the cost function, and
+    :math:`D = \mathrm{diag}(v^{1/2})`, where :math:`v` is defined as
+
+    .. math::
+
+        v_i = \begin{cases}
+            ub_i - x_i & g_i < 0 \text{ and } ub_i < \infty \\
+            x_i - lb_i & g_i > 0 \text{ and } lb_i > -\infty \\
+            1 & \text{otherwise}
+        \end{cases}
+
+    This system constitutes first-order optimality condition for the original
+    optimization problem. The algorithm computes a Newton step for this system
+    and then performs line search along it with possible reflections from the
+    bounds.
+
+    References
+    ----------
+    .. [1] T. F. Coleman and Y. Li. "A Reflective Newton Method for
+           Minimizing a Quadratic Function Subject to Bounds on Some of the
+           Variables," SIAM Journal on Optimization, Vol. 6, Number 4,
+           pp. 1040-1058, 1996.
+
+    Examples
+    --------
+    In this example a problem with large sparse matrix is solved.
+
+    >>> from scipy.sparse import rand
+    >>> from scipy.optimize import lsq_linear
+    ...
+    >>> np.random.seed(0)
+    ...
+    >>> m = 20000
+    >>> n = 10000
+    ...
+    >>> A = rand(m, n, density=1e-4)
+    >>> b = np.random.randn(m)
+    ...
+    >>> lb = np.random.randn(n)
+    >>> ub = lb + 1
+    ...
+    >>> res = lsq_linear(A, b, lb, ub, verbose=1)
+    Number of iterations: 14, initial cost: 1.5166e+04, final cost 1.1112e+04,
+    first-order optimality 5.07e-11.
```

Comment:
No, it's just a mistake) I dropped the first line for some reasons, Will fix it soon.